File: Flowsource-templates/node-gcp-ci-template/skeleton/src/exception/ResourceNotFoundException.ts
import { HttpException, HttpStatus } from '@nestjs/common';

export class ResourceNotFoundException extends HttpException {
  constructor(message: string, cause?: any);
  constructor(invalidId: number);
  constructor(messageOrId: string | number, cause?: any) {
    if (typeof messageOrId === 'number') {
      super(`Resource with id ${messageOrId} not found`, HttpStatus.NOT_FOUND);
    } else {
      super(messageOrId, HttpStatus.NOT_FOUND);
    }
  }
}

File: Flowsource-templates/python-github-aws-ci-pipeline-template/skeleton/app/exception/resource_not_found_exception.py
class ResourceNotFoundException(Exception):
    """Exception raised when a resource is not found."""

    def __init__(self, message, cause=None):
        super().__init__(message)
        if cause:
            self.__cause__ = cause

    @classmethod
    def from_invalid_id(cls, invalid_id):
        return cls(f"Resource with id {invalid_id} not found")
